airflow/models/event_note.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from sqlalchemy import Column, Integer, String, Text
+from airflow.models.base import COLLATION_ARGS, ID_LEN, Base
+from airflow.utils import timezone
+from airflow.utils.sqlalchemy import UtcDateTime
+
+log = logging.getLogger(__name__)
+
+
+class EventNote(Base):
+    """Model that stores a note for an event"""
+
+    __tablename__ = "event_note"
+
+    id = Column(Integer, primary_key=True)
+
+    dag_id = Column(String(ID_LEN, **COLLATION_ARGS))
+    task_id = Column(String(ID_LEN), nullable=True)
+    execution_date = Column(UtcDateTime, nullable=True)
```

Comment:
Better to point to run_id. But I actually feel this is restricting the usage too narrowly; a better way might be to have a field called `type` or something, and an identifier field to store a JSON blob, and store an identifier that makes sense for each event type (e.g. the  primary of a DAG run, a task isntance, or something else).